### PR TITLE
Add augment bounding box support for RandomCrop

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
randomcrop_demo_bbox_2
random_crop_bbox_demo1

### CODE REVIEW

File: keras_cv/layers/preprocessing/random_crop.py [hunk 1]
```diff
@@ -80,14 +81,47 @@ def augment_image(self, image, transformation, **kwargs):
...
+            target="xyxy",
+            images=image,
+        )
+        image_shape = tf.shape(image)
+        h_diff = image_shape[H_AXIS] - self.height
+        w_diff = image_shape[W_AXIS] - self.width
+        bounding_boxes = tf.cond(
+            tf.reduce_all((h_diff >= 0, w_diff >= 0)),
+            lambda: self._crop_bounding_boxes(image, bounding_boxes, transformation),
+            lambda: self._resize_bounding_boxes(
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_crop.py:111]: Are we always resizing down? If not, we may need to clip these to the image (using keras_cv.bounding_box.clip_to_image)

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks Divya!

File: examples/layers/preprocessing/bounding_box/random_crop_demo.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import demo_utils
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing import RandomCrop
+
+IMG_SIZE = (256, 256)
+BATCH_SIZE = 9
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply examples/layers/preprocessing/bounding_box/random_crop_demo.py:27]: done

File: keras_cv/layers/preprocessing/random_crop.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_cv/layers/preprocessing/random_crop.py:111]: Ahh! just figured I can provide image_shape to clip_to_image. I was stuck here thinking it only crops to original image size.
I have updated the changes.

**Review by ianstenbit — APPROVED** [review]

**Pull request merged**